### PR TITLE
add mobile workflow template

### PR DESCRIPTION
**Pull request by StephenHodgson**

### CODE REVIEW

File: .github/workflows/mobile-build.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        os: [windows-latest, macos-13]
+        include:
+          - os: windows-latest
+            build-target: Android
+            build-args: ''
+          - os: macos-13
+            build-target: iOS
+            build-args: ' -enableAppleAutomaticSigning'
+    steps:
+      - uses: actions/checkout@v4
```
Comment (afarchy) [inline .github/workflows/mobile-build.yml:37]: Probably should leave commends on each step as we do in the other one.

File: .github/workflows/mobile-build.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          log-name: 'project-validation'
+          args: '-quit -batchmode -executeMethod Utilities.Editor.BuildPipeline.UnityPlayerBuildTools.ValidateProject'
+      - uses: buildalon/unity-action@v1
+        name: '${{ matrix.build-target }}-Build'
+        with:
+          log-name: '${{ matrix.build-target }}-Build'
+          build-target: '${{ matrix.build-target }}'
+          args: '-quit -batchmode -executeMethod Utilities.Editor.BuildPipeline.UnityPlayerBuildTools.StartCommandLineBuild${{ matrix.build-args }}'
+      - uses: buildalon/unity-xcode-builder@v1
+        if: matrix.os == 'macos-13' && success()
```
Comment (afarchy) [inline .github/workflows/mobile-build.yml:62]: check the build-target instead so we don't have to replace macos-13 here for buildalon

File: .github/workflows/mobile-build.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ ( github.event_name == 'pull_request' || github.event.action == 'synchronize' ) }}
+jobs:
+  build:
+    env:
+      UNITY_PROJECT_PATH: '' # set by unity-setup action
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest, macos-13]
```
Comment (afarchy) [inline .github/workflows/mobile-build.yml:28]: these templates should probably use buildalon by default.

Comment (StephenHodgson) [reply .github/workflows/mobile-build.yml:28]: No these are specifically for github hosted runners

**Pull request merged**